wgengine/magicsock/magicsock.go

```diff
@@ -3023,9 +3023,13 @@ type RebindingUDPConn struct {
 
 	mu    sync.Mutex // held while changing pconn (and pconnAtomic)
 	pconn nettype.PacketConn
+	// laddr is laziliy memoized for frequent access from disco, cleared when
+	// the pconn is updated, and populated by the next call to LocalAddr.
+	laddr *net.UDPAddr
```

Comment:
store a netip.AddrPort instead? and then don't worry about the defensive copy stuff

Reply:
Oh, PacketConn.LocalAddr() actually has a very loose specification. Sadly netip.AddrPort does quite meet the specification so I can't just return that.

I can just add a .Port() extension here instead.

Reply:
done, PTAL.